lib/src/data/environment_base.dart

```diff
@@ -0,0 +1,9 @@
+abstract class EnvironmentBase {
+  /// The name of the environment.
+  final String name = '';
+
+  /// Whether or not to report-only Content Security Policy violations.
+  final bool shouldEnforceCSP = false;
+
+  final Map<String, dynamic> versionFileEntries = const {};
```

Comment:
Don't set default values when you define an interface. They are basically unused. Only expose getters/methods